src/nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -928,7 +950,59 @@ def apply(
         self.set_backend_entity(model)
 
         # Get processed weight compression parameters ready for compression
-        all_weight_params, statistics = self.get_weight_compression_parameters(model, graph, statistic_points, dataset)
+        all_weight_params, ratio_defining_params, skipped_weight_params = self.get_weight_compression_parameters(
+            model, graph
+        )
+        return self.apply_with_parameters(
+            model,
+            graph,
+            dataset,
+            statistic_points,
+            all_weight_params,
+            ratio_defining_params,
+            skipped_weight_params,
+        )
+
+    def apply_with_parameters(
+        self,
+        model: TModel,
+        graph: NNCFGraph,
+        dataset: Dataset,
+        statistic_points: StatisticPointsContainer,
+        all_weight_params: list[WeightCompressionParameters],
+        ratio_defining_params: list[WeightCompressionParameters],
+        skipped_weight_params: list[WeightCompressionParameters],
+    ) -> TModel:
+        """
+        Applies the Weight Compression algorithm using precomputed parameters and optional
+        algorithms (AWQ, GPTQ, scale estimation, LoRA correction). The method collects
+        statistics, configures the weight compression parameters for mixed precision algorithm,
+        and performs the model transformation with appropriate decompression operations
+
+        :param model: Backend-specific model to be compressed.
+        :param graph: NNCFGraph instance.
+        :param dataset: Dataset to collect statistics.
+        :param statistic_points: Statistics points object.
+        :param all_weight_params: List of all weight parameters.
+        :param ratio_defining_params: Subset of all_weight_params that determine mixed-precision ratios.
+        :param skipped_weight_params: List of parameters corresponding to weights intentionally skipped
+            from compression (e.g., due to ignored scopes or group size adjustments).
+        :return: Transformed model with compressed weights and inserted backend-specific decompressor.
+        """
+        # Collect statistics for the weights compression
+        statistics, statistic_points = self._collect_statistics_and_statistic_points(
+            model, graph, statistic_points, dataset, ratio_defining_params, all_weight_params
+        )
+        # Set weight compression configuration
+        self._set_weight_compression_config(ratio_defining_params, model, graph, statistic_points)
+
+        # Print statistics
+        nncf_logger.info(
+            self._get_bitwidth_distribution_str(all_weight_params, ratio_defining_params, skipped_weight_params)
+        )
+
+        # Filter all_weight_params and by excluding nodes that should remain in their original floating-point precision
+        all_weight_params = list(filter(lambda w_params: w_params.compression_config is not None, all_weight_params))
```

Comment:
```suggestion
        # Filter all_weight_params by excluding nodes that should remain in their original floating-point precision
        all_weight_params = [w_params for w_params in all_weight_params if w_params.compression_config is not None]
```
AFAIK, list comprehension in this case is a bit more efficient.